### PR TITLE
chore(shorebird_cli): release 1.0.3

### PR DESCRIPTION
**Pull request by bryanoltman**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 This section contains past updates we've sent to customers via Discord.
 
+## 1.0.3 (April 25, 2024)
+
+- 🐦 Support for Flutter 3.19.6
+- 🖨️ Improve logging for patch commands to include Flutter version being used to build
+- 💾 Improve consistency of Shorebird artifact downloads
+- ✅ Add validation that shorebird.yaml is present in pubspec.yaml assets section
+- 🌌 Improved handling of multi-dimensional Android flavors
```
Comment (erickzanardo) [inline RELEASE_NOTES.md:11]: should we mention the linker debug data too? I feel that is quite a cool feature?

Comment (bryanoltman) [reply RELEASE_NOTES.md:11]: Sure! Throw a suggestion in there :)

Comment (erickzanardo) [reply RELEASE_NOTES.md:11]:
```suggestion
- 🌌 Improved handling of multi-dimensional Android flavors
- 🔎 New option on `patch ios` command to collect debug data about the linking
```

Comment (bryanoltman) [reply RELEASE_NOTES.md:11]: Added!

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**